### ISSUE TITLE
Mac: FirstFrameReady happens while still showing a garbage image from the previous video

### ISSUE THREAD

**Issue by TheloTheGreat** [RenderHeads/UnityPlugin-AVProVideo#659]
On Mac (but not Windows), when opening a new video after playing an old video, the FirstFrameReady event fires while we still see the old video on the MediaPlayer's display (DisplayUGUI, for instance). The FirstFrameReady should only fire when a valid image from the new video really is on-screen.

To repro, on Mac AVPro 2.0.8 in Demo_MediaPlayer, in MediaPlayerUI.cs:

Add this to Start():
```
if (_mediaPlayer) _mediaPlayer.Events.AddListener(OnMediaPlayerEvent);
```

Add this new method:
```
public void OnMediaPlayerEvent(MediaPlayer mp, MediaPlayerEvent.EventType type, ErrorCode errorCode) {
	if (type == MediaPlayerEvent.EventType.FirstFrameReady) {
		Debug.LogError("First frame ready"); // Auto-pause the editor
	}
}
```

Add this to Update:
```
if (Input.GetKeyDown(KeyCode.Q)) {
	_mediaPlayer.OpenMedia(MediaPathType.RelativeToProjectFolder, "Assets/StreamingAssets/AVProVideoSamples/BigBuckBunny-360p30-H264.mp4");
}
if (Input.GetKeyDown(KeyCode.W)) {
	_mediaPlayer.OpenMedia(MediaPathType.RelativeToProjectFolder, "Assets/StreamingAssets/AVProVideoSamples/RenderHeads-1080p30-H264.mp4");
}
```

Then launch the scene, manually unpausing the editor once to allow the initial video to start playing a bit. Press the Q or the W key while the video is playing to open a new video, and the editor will pause when the FirstFrameReady event happens.

I expect the editor to pause on the first frame of the new video, but instead it pauses on a frame of the old video, which is incorrect.

(In practice, it means I cannot use FirstFrameReady in my UI to start showing a video, because that will show a flicker of incorrect video before showing the intended one.)

**Comment by MorrisRH** [comment]
A fix will be in the next release.

**Comment by TheloTheGreat** [comment]
Fix confirmed in 2.0.9, thanks.
